app/controllers/merge_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class MergeController < ApplicationController
+  before_action :require_admin
+
+  def preview
+    @from_teacher = Teacher.find(params[:from])
+    @into_teacher = Teacher.find(params[:into])
+    @result_teacher = merge_teachers(@from_teacher, @into_teacher)
+  end
+
+  def execute
+    @from_teacher = Teacher.find(params[:from])
+    @into_teacher = Teacher.find(params[:into])
+    @merged_teacher = merge_teachers(@from_teacher, @into_teacher)
+
+    merged_attributes = @merged_teacher.attributes.except("id")
+    @into_teacher.update!(merged_attributes)
+    @from_teacher.destroy
+    redirect_to teachers_path, notice: "Teachers merged successfully."
+  end
+
+  private
+  # Returns a merged teacher without saving it to the database.
+  # Rendered in the preview, and only saved to the DB in a call to merge.
+  def merge_teachers(from_teacher, into_teacher)
```

Comment:
This should be a method in lib/ or somewhere else, but not gonna move it for now